snapcraft/plugins/go.py

```diff
@@ -163,17 +151,25 @@ def clean_build(self):
         if os.path.isdir(self._gopath_pkg):
             shutil.rmtree(self._gopath_pkg)
 
-    def _local_build(self):
-        if self.options.go_importpath:
-            go_package = self.options.go_importpath
-        else:
-            go_package = os.path.basename(os.path.abspath(self.options.source))
-        self._run(['go', 'install', './{}/...'.format(go_package)])
+    def _run(self, cmd, **kwargs):
+        env = self._build_environment()
+        return self.run(cmd, cwd=self._gopath_src, env=env, **kwargs)
 
-    def _remote_build(self):
-        for go_package in self.options.go_packages:
-            self._run(['go', 'install', go_package])
+    def _build_environment(self):
+        env = os.environ.copy()
+        env['GOPATH'] = self._gopath
 
-    def _run(self, cmd, **kwargs):
-        cmd = ['env', 'GOPATH={}'.format(self._gopath)] + cmd
-        return self.run(cmd, cwd=self._gopath_src, **kwargs)
+        # Add $GOPATH/bin so godeps is actually callable.
+        env['PATH'] = '{}:{}'.format(
+            os.path.join(self._gopath, 'bin'), env.get('PATH', ''))
+
+        include_paths = []
+        for root in [self.installdir, self.project.stage_dir]:
+            include_paths.extend(
+                common.get_library_paths(root, self.project.arch_triplet))
+
+        flags = common.combine_paths(include_paths, '-L', ' ')
+        env['CGO_LDFLAGS'] = '{} {} {}'.format(
+            env.get('CGO_LDFLAGS', ''), flags, env.get('LDFLAGS', ''))
```

Comment:
But wouldn't there be some spurious weird empty spaces if the env.gets return ''?
`' '.join(filter(None, [env.get('CGO_LDFLAGS'), flags, env.get('LDFLAGS')]))`

Of course, I don't really care. If you prefer yours, I'm ok.